src/test/java/org/folio/rest/api/DereferencedItemStorageTest.java

```diff
@@ -0,0 +1,266 @@
+package org.folio.rest.api;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.folio.rest.api.StorageTestSuite.TENANT_ID;
+import static org.folio.rest.support.ResponseHandler.json;
+import static org.folio.rest.support.http.InterfaceUrls.holdingsStorageUrl;
+import static org.folio.rest.support.http.InterfaceUrls.instancesStorageUrl;
+import static org.folio.rest.support.http.InterfaceUrls.itemsStorageUrl;
+import static org.folio.rest.support.http.InterfaceUrls.dereferencedItemStorage;
+import static org.folio.util.StringUtil.urlEncode;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertNull;
+
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.folio.rest.jaxrs.model.DereferencedItem;
+import org.folio.rest.jaxrs.model.DereferencedItems;
+import org.folio.rest.support.Response;
+import org.folio.rest.support.ResponseHandler;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+
+import io.vertx.core.json.JsonObject;
+import junitparams.JUnitParamsRunner;
+import lombok.SneakyThrows;
+
+
+public class DereferencedItemStorageTest extends TestBaseWithInventoryUtil {
+  private static final UUID smallAngryPlanetId = UUID.randomUUID();
+  private static final UUID uprootedId = UUID.randomUUID();
+
+
+  @BeforeClass
+  public static void beforeTests() throws InterruptedException, ExecutionException, TimeoutException {
+    StorageTestSuite.deleteAll(itemsStorageUrl(""));
+    StorageTestSuite.deleteAll(holdingsStorageUrl(""));
+    StorageTestSuite.deleteAll(instancesStorageUrl(""));
+
+    UUID holdingsRecordId = createInstanceAndHolding(mainLibraryLocationId);
+
+    JsonObject smallAngryPlanet = smallAngryPlanet(smallAngryPlanetId, holdingsRecordId);
+    JsonObject nod = nod(UUID.randomUUID(), holdingsRecordId);
+    JsonObject uprooted = uprooted(uprootedId, holdingsRecordId);
+
+    postItem(smallAngryPlanet);
+    postItem(nod);
+    postItem(uprooted);
+  }
+
+  @AfterClass
+  public static void cleanUpDatabase() {
+    StorageTestSuite.deleteAll(itemsStorageUrl(""));
+    StorageTestSuite.deleteAll(holdingsStorageUrl(""));
+    StorageTestSuite.deleteAll(instancesStorageUrl(""));
+  }
+
+  
+  public void CanGetRecordByCQLSearch() {
+    String queryString = "barcode=036000291452";
```

Comment:
If this is intended to demonstrate an exact match on barcode (which I believe is the use case), the relation should be `==`

Reply:
I don't follow-this is an assignment, I'm putting the query string for the search into a variable.

Reply:
Fixed.